plugins/fsm/src/__model__.ts

```diff
@@ -0,0 +1,158 @@
+// Copyright © 2020, 2021 Anticrm Platform Contributors.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import core, { ArrayOf$, Builder, Class$, Mixin$, Prop, RefTo$, InstanceOf$ } from '@anticrm/model'
+import { CORE_CLASS_STRING, Ref, Class, Doc } from '@anticrm/core'
+import { Application } from '@anticrm/domains'
+import { TDoc, TEmb } from '@anticrm/model/src/__model__'
+import { UX } from '@anticrm/presentation/src/__model__'
+import { IntlString } from '@anticrm/platform-i18n'
+
+import fsmPlugin, { FSM, Transition, State, WithFSM, WithState } from '.'
+
+const FSMDomain = 'fsm'
+
+@UX('FSM' as IntlString)
+@Class$(fsmPlugin.class.FSM, core.class.Doc, FSMDomain)
+export class TFSM extends TDoc implements FSM {
+  @UX('Name' as IntlString)
+  @Prop(CORE_CLASS_STRING)
+  name!: string
+
+  @RefTo$(core.class.Application)
+  application!: Ref<Application>
+
+  @UX('Name' as IntlString)
+  @ArrayOf$()
+  @InstanceOf$(fsmPlugin.class.Transition)
+  transitions!: Transition[]
```

Comment:
I'm agree with @z2sx  we should try not use embedded objects here and try to describe transitions as separate top level objects.